census_extractomatic/api.py

```diff
@@ -2009,6 +2021,303 @@ def table_geo_comparison_rowcount(table_id):
+        since, generally, simpler, more complete tables are more useful. This
+        function selects the most relevant table based on the hierarchy above.
+
+        Table IDs are in the format [B/C]#####[A-I]. The first character is 
+        'B' or 'C', followed by five digits (the tabulation code), optionally
+        ending with a character representing that this is a race iteration. 
+        If any iteration is present, all of them are (e.g., if B10001A is 
+        present, so are B10001B, ... , B10001I.)
+        """
+
+        tabulation_code = tables[0][1:6]
+
+        # 'C' table with no iterations, e.g., C10001
+        if 'C' + tabulation_code in tables:
+            return 'C' + tabulation_code
+
+        # 'B' table with no iterations, e.g., B10001
+        if 'B' + tabulation_code in tables:
+            return 'B' + tabulation_code
+
+        # 'C' table with iterations, choosing 'A' iteration, e.g., C10001A
+        if 'C' + tabulation_code + 'A' in tables:
+            return 'C' + tabulation_code + 'A'
+
+        # 'B' table with iterations, choosing 'A' iteration, e.g., B10001A
+        if 'B' + tabulation_code + 'A' in tables:
+            return 'B' + tabulation_code + 'A'
+
+        else:
+            return ''
+
+    def process_result(row):
+        """ Converts a SQLAlchemy RowProxy to a dictionary. 
+
+        params: row - row object returned from a query
+        return: dictionary with either profile or table attributes """
+
+        row = dict(row)
+
+        if row['type'] == 'profile':
+            result = {
+                'type': 'profile',
+                'full_geoid': row['full_geoid'],
+                'full_name': row['display_name'],
+                'sumlevel': row['sumlevel'],
+                'sumlevel_name': row['sumlevel_name'] if row['sumlevel_name'] else '',
+                'url': build_profile_url(row['display_name'], row['full_geoid']),
+                'relevance': row['relevance'] #TODO remove this
+            }
+
+        if row['type'] == 'table':
+            table_id = choose_table(row['tables'].split())
+
+            result = {
+                'type': 'table',
+                'table_id': table_id,
+                'tabulation_code': row['tabulation_code'],
+                'table_name': row['table_title'],
+                'simple_table_name': row['simple_table_title'],
+                'topics': row['topics'].split(', '),
+                'unique_key': row['tabulation_code'],
+                'subtables': row['tables'].split(),
+                'url': build_table_url(table_id),
+                'relevance': row['relevance'] #TODO remove this
+
+            }
+
+        if row['type'] == 'topic':
+            result = {
+                'type': 'topic',
+                'topic_name': row['topic_name'],
+                'url': row['url'],
+                'relevance': row['relevance'] #TODO remove this
+            }
+
+        return result
+
+    def slugify(name):
```

Comment:
I'm pretty sure we already have a library for slugify that's used elsewhere in the code. Can you check for it and use that instead? (I might be mis-remembering or it might just be in the Django app, in which case nevermind)